diff --git a/core/core.el b/core/core.el
index 07254e602..3df3ce53b 100644
--- a/core/core.el
+++ b/core/core.el
@@ -143,7 +143,9 @@ envvar will enable this at startup.")
 (defconst doom-local-dir
   (if-let (localdir (getenv-internal "DOOMLOCALDIR"))
       (expand-file-name (file-name-as-directory localdir))
-    (concat doom-emacs-dir ".local/"))
+    (expand-file-name "doom-local/"
+                      (or (getenv-internal "XDG_STATE_HOME")
+                          "~/.local/state")))
   "Root directory for local storage.
 
 Use this as a storage location for this system's installation of Doom Emacs.
@@ -151,15 +153,22 @@ Use this as a storage location for this system's installation of Doom Emacs.
 These files should not be shared across systems. By default, it is used by
 `doom-etc-dir' and `doom-cache-dir'. Must end with a slash.")
 
+(defconst doom-state-dir
+  (if-let (localdir (getenv-internal "DOOMLOCALDIR"))
+      (concat (expand-file-name (file-name-as-directory localdir)) "state/")
+    (expand-file-name "doom/"
+                      (or (getenv-internal "XDG_STATE_HOME")
+                          "~/.local/state"))))
+
 ;; DEPRECATED
-(defconst doom-etc-dir (concat doom-local-dir "etc/")
+(defconst doom-etc-dir (concat doom-state-dir "etc/")
   "Directory for non-volatile local storage.
 
 Use this for files that don't change much, like server binaries, external
 dependencies or long-term shared data. Must end with a slash.")
 
 ;; DEPRECATED
-(defconst doom-cache-dir (concat doom-local-dir "cache/")
+(defconst doom-cache-dir (concat doom-state-dir "cache/")
   "Directory for volatile local storage.
 
 Use this for files that change often, like cache files. Must end with a slash.")
@@ -237,7 +246,7 @@ users).")
 (when NATIVECOMP
   ;; Don't store eln files in ~/.emacs.d/eln-cache (where they can easily be
   ;; deleted by 'doom upgrade').
-  (add-to-list 'native-comp-eln-load-path (concat doom-cache-dir "eln/"))
+  (add-to-list 'native-comp-eln-load-path (concat doom-local-dir "eln/"))
 
   (with-eval-after-load 'comp
     ;; HACK Disable native-compilation for some troublesome packages
@@ -271,7 +280,7 @@ users).")
 
 ;; Allow the user to store custom.el-saved settings and themes in their Doom
 ;; config (e.g. ~/.doom.d/).
-(setq custom-file (expand-file-name "custom.el" doom-private-dir))
+(setq custom-file (expand-file-name "custom.el" doom-state-dir))
 
 (define-advice en/disable-command (:around (fn &rest args) write-to-data-dir)
   "Write saved safe-local-variables to `custom-file' instead.

diff --git a/modules/app/rss/config.el b/modules/app/rss/config.el
index a8078d18f..075a5c8a7 100644
--- a/modules/app/rss/config.el
+++ b/modules/app/rss/config.el
@@ -20,8 +20,8 @@ easier to scroll through.")
 (use-package! elfeed
   :commands elfeed
   :init
-  (setq elfeed-db-directory (concat doom-local-dir "elfeed/db/")
-        elfeed-enclosure-default-dir (concat doom-local-dir "elfeed/enclosures/"))
+  (setq elfeed-db-directory (concat doom-cache-dir "elfeed/db/")
+        elfeed-enclosure-default-dir (concat doom-cache-dir "elfeed/enclosures/"))
   :config
   (setq elfeed-search-filter "@2-week-ago "
         elfeed-show-entry-switch #'pop-to-buffer
